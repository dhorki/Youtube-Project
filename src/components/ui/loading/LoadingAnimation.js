import React from 'react';
import { StyledLoadingAnimation } from '../../../styles/components/ui/loading/LoadingAnimation';

const loadingAnimation = {
  src: 'https://i.pinimg.com/originals/d4/2b/b2/d42bb2f00e1042c990b25f90ada9b0ab.gif',
};

export const LoadingAnimation = () => {
  return (
    <StyledLoadingAnimation>
      <img src={loadingAnimation.src} alt="loading" />
    </StyledLoadingAnimation>
  );
};
