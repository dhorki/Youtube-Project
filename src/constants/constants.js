export const styles = {
  colors: {
    light: {
      appIcon: '#ff0200',
      background: '#F9F9F9',
      barBackground: '#ffffff',
      primary: '#030303',
      secondary: '#606060',
      linkActive: '#ff0200',
      linkHoverBackground: '#ececec',
      inputBackground: '#ffffff',
      inputBorder: '#cccccc',
      searchButtonBackground: '#f8f8f8',
      // google button
      white: '#fff',
      googleBlue: '#4285f4',
      buttonActiveBlue: '#1669f2',
    },
    dark: {
      appIcon: '#ff0200',
      background: '#181818',
      barBackground: '#212121',
      primary: '#ffffff',
      secondary: '#aaaaaa',
      linkActive: '#ff0200',
      linkHoverBackground: '#303030',
      inputBackground: '#121212',
      inputBorder: '#303030',
      searchButtonBackground: '#323232',
      // google button
      white: '#fff',
      googleBlue: '#4285f4',
      buttonActiveBlue: '#1669f2',
    },
  },
  props: {
    transitionTime: '0.4s',
    sidebarWidth: '150px',
  },
};

export const screenSizes = {
  xsmall: '675px',
  small: '830px',
  mid: '1200px',
};

export const mainViewTypes = {
  searchGallery: 'searchGallery',
  favoritesGallery: 'favoritesGallery',
  videoDetails: 'videoDetails',
};

export const environmentActions = {
  setTheme: 'setTheme',
  setSidebarShow: 'setSidebarShow',
  setSearchQuery: 'setSearchQuery',
  setModalShow: 'setModalShow',
  setUser: 'setUser',
};

export const modalTypes = {
  login: 'login',
  register: 'register',
  resetPassword: 'reset-password',
  none: 'none',
};

export const resetSearchQueryAction = {
  type: environmentActions.setSearchQuery,
  payload: {
    q: '',
  },
};

export const hideModalAction = {
  type: environmentActions.setModalShow,
  payload: {
    modalShow: modalTypes.none,
  },
};
