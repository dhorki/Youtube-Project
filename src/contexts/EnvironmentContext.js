import { createContext } from 'react';

export const EnvironmentContext = createContext();
