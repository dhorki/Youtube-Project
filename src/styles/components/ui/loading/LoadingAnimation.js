import styled from 'styled-components';

export const StyledLoadingAnimation = styled.div`
  width: 200px;
  height: 200px;
  overflow: hidden;

  img {
    position: relative;
    top: -200px;
    left: -300px;
    text-align: center;
    height: 600px;
    width: 800px;
    /* background-image: url($loadingUrl); */
    background-position: center;
    mix-blend-mode: exclusion;
    filter: invert(100%);
  }
`;
