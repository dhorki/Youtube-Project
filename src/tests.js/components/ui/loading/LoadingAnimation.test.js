import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { styles } from '../../../../constants/constants';
import { EnvironmentContext } from '../../../../contexts/EnvironmentContext';
import { LoadingAnimation } from '../../../../components/ui/loading/LoadingAnimation';

describe('LoadingAnimation', () => {
  const environment = {
    theme: styles.colors.light,
    sidebarShow: false,
    q: '',
  };

  it('should render correctly', () => {
    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <LoadingAnimation />
      </EnvironmentContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
