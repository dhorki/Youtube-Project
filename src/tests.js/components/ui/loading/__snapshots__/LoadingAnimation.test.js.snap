// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAnimation should render correctly 1`] = `
<div
  className="sc-bczRLJ ihClzx"
>
  <img
    alt="loading"
    src="https://i.pinimg.com/originals/d4/2b/b2/d42bb2f00e1042c990b25f90ada9b0ab.gif"
  />
</div>
`;
